models/demos/t3000/llama2_70b/demo/eval_t3000.py

```diff
@@ -0,0 +1,222 @@
+    num_samples: int = 128
+    perplexity_score: float = 5.4
+
+
+def main(args, eval_data_args):
+    # Set random reproducible seed
+    torch.manual_seed(0)
+    model_args = args.model
+    tt_args = args.tt
+    data_args = args.data
+
+    generator = build_generator(model_args, tt_args)
+
+    # Load the model and tokenizer
+    model, tokenizer = generator.model, generator.tokenizer
+
+    # Dataset preparation
+    dataset = datasets.load_dataset(
+        eval_data_args.dataset, eval_data_args.config, split=eval_data_args.split, ignore_verifications=True
+    )
+    text = wikitext_detokenizer("\n".join(dataset["text"]))
+    encodings = tokenizer.encode(text, bos=True, eos=False)  # not prepending bos
+
+    # args for perplexity calculation
+    max_length = model_args.max_seq_len
+    stride = eval_data_args.stride if eval_data_args.stride else max_length
+    assert stride <= max_length, "stride cannot be larger than max_length"
+    seq_len = eval_data_args.sample_len
+    num_samples = eval_data_args.num_samples
+    max_batch_size = model_args.max_batch_size
+    assert num_samples > 0, "num_samples must be greater than 0"
+    assert seq_len + (num_samples - 1) * stride <= len(encodings), (
+        "total length of token decoded must be less than the length of the dataset, \
+        the maximum allowed num_samples is: %d"
+        % (len(encodings) - seq_len)
+        // stride
+        + 1
+    )
+
+    perplexity = calculate_perplexity(
+        model_args,
+        tt_args,
+        data_args,
+        eval_data_args,
+        model,
+        tokenizer,
+        seq_len,
+        max_length,
+        stride,
+        encodings,
+        num_samples,
+        max_batch_size,
+    )
+
+    base_path = "models/demos/t3000/llama2_70b/scripts/llama_perplexity_runs/"
+    os.makedirs(base_path, exist_ok=True)
+
+    # Get current timestamp
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+
+    # Dump perplexity and max_seq_len to a JSON file with timestamp and max_length in the file name
+    filename = os.path.join(
+        base_path,
+        f"perplexity_{model_args.llama_version}_{model_args.implementation}_{model_args.num_layers}L_"
+        f"{eval_data_args.sample_len}_{data_args.max_output_tokens}_{timestamp}.json",
+    )
+    result = {
+        "model": model_args.llama_version,
+        "perplexity": perplexity.item(),
+        "seq_len": eval_data_args.sample_len,
+        "gen_length": data_args.max_output_tokens,
+    }
+    # create folder if it doesn't exist
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+    with open(filename, "w") as f:
+        json.dump(result, f)
+
+    logger.info("Perplexity: %f" % perplexity)
+    if perplexity < eval_data_args.perplexity_score:
+        logger.info("Perplexity is less than the threshold")
+    else:
+        assert False, "Perplexity is greater than the threshold"
+
+    return perplexity
+
+
+@pytest.mark.timeout(240000)
```

Comment:
Is there a reason to have a 66 hour timeout?
We already implement a high level timeout on the script level here
```
  WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto models/demos/t3000/llama2_70b/demo/eval_t3000.py --timeout=7200 ; fail+=$?
```

which will disocver and hung tests and continue them for workflows.

Reply:
got it will remove it from all our CI tests, this was for local testing when sometime it would go over night

Reply:
Timeout has been removed @ttmchiou could you please review thanks!

Reply:
@ttmchiou could you please take a look thanks